msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/ManagedIdentityRequest.java

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+class ManagedIdentityRequest extends MsalRequest {
+
+    URI baseEndpoint;
+
+    HttpMethod method;
+
+    Map<String, String> headers;
+
+    Map<String, String> bodyParameters;
+
+    Map<String, String> queryParameters;
+
+    public ManagedIdentityRequest(ManagedIdentityApplication managedIdentityApplication, RequestContext requestContext){
+        super(managedIdentityApplication, requestContext);
+    }
+
+    public URL computeURI() throws URISyntaxException {
+        String endpoint = this.appendQueryParametersToBaseEndpoint();
+        try {
+            return new URL(endpoint);
+        } catch (MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String appendQueryParametersToBaseEndpoint(){
+        StringBuilder stringBuilder = new StringBuilder(baseEndpoint.toString());
```

Comment:
Do you need to do this manually ? Isn't there URL / URI methods that help with this?

Reply:
We use a helper method from Nimbus called `URLUtils.serializeParameters()` to serialize parameters into a string in a few places (see [here](https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/4dc7a5e3c29e3995c585648b86cb3a04aef3c39f/msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/TokenRequestExecutor.java#L79) for an example of serializing query params), but it takes in a `Map<String, List<String>>` instead of a `Map<String, String>`. This means we'd need to refactor every instance of `queryParameters.put("key", "value")` to `queryParameters.put("key", Collections.singleton("value"))`, which is clunky but we actually do it quite a lot in the library already...